### PR TITLE
feat: fetch secrets from OneGraph as-needed

### PR DESCRIPTION
**Pull request by dwwoelfel**
Fetches secrets on each usage of `withSecrets` using the `ONEGRAPH_AUTHLIFY_TOKEN`. This works around the problem of needing to redeploy to update the secrets

- [X] I have read the [contribution guidelines](../blob/master/CONTRIBUTING.md).
- [x] The status checks are successful (continuous integration). Those can be seen below.

### CODE REVIEW

File: src/lib/secrets.js [hunk 1]
```diff
@@ -1,33 +1,149 @@
...
       }
     }
-    return secrets
-  }, {})
+  }`
+
+  const body = JSON.stringify({ query: doc })
+
+  // eslint-disable-next-line node/no-unsupported-features/node-builtins
+  const result = await oneGraphRequest(secretToken, new TextEncoder().encode(body))
```
Comment (ascorbic) [inline src/lib/secrets.js:130]: `encode()` returns a `Uint8Array`, but `ClientRequest` expects a `Buffer`. Newer versions of Node seem to be forgiving about this, but Node 12 fails. I'd suggest passing `body` to `oneGraphRequest`, and then call `Buffer.from(new TextEncoder().encode(requestBody))` in there.

File: src/lib/secrets.d.ts [hunk 2]
```diff
@@ -2,14 +2,33 @@ import { Context } from '../function/context'
...
+    }
+  }> | null
+}
+
 export type Services = typeof services
 
 export type ServiceKey = keyof Services
 
-export type ServiceTokens<T extends ServiceKey> = Services[T]['tokens']
+export type ServiceTokens = Service
```
Comment (ascorbic) [inline src/lib/secrets.d.ts:28]: I don't think we need to maintain backward-compatibility in a prerelease. Go ahead an remove this.

File: src/lib/secrets.d.ts [hunk 3]
```diff
@@ -2,14 +2,33 @@ import { Context } from '../function/context'
 import { Handler } from '../function/handler'
 import * as services from './services.json'
 
+export type Service = {
+  friendlyServiceName: string
+  service: string
+  isLoggedIn: boolean
+  bearerToken: string | null
+  grantedScopes: Array<{
```
Comment (ascorbic) [inline src/lib/secrets.d.ts:10]: It would be better to define a `Scope`, rather than doing it inline.

File: src/lib/services.json [hunk 4]
```diff
@@ -1,17 +1,6 @@
...
-    "tokens": {
-      "token": "ONEGRAPH_SPOTIFY_TOKEN"
-    }
-  },
-  "salesforce": {
-    "tokens": {
-      "token": "ONEGRAPH_SALESFORCE_TOKEN"
-    }
-  }
+  "gitHub": null,
```
Comment (ascorbic) [inline src/lib/services.json:2]: There's not really much point doing this in a JSON file at this point. Just define the type in the d.ts.

File: src/lib/secrets.js [hunk 5]
```diff
@@ -1,33 +1,149 @@
...
+
+    req.write(requestBody)
+
+    req.end()
+  })
+}
+
+const formatSecrets = (result) => {
+  const services =
+    result.data && result.data.me && result.data.me.serviceMetadata && result.data.me.serviceMetadata.loggedInServices
```
Comment (ascorbic) [inline src/lib/secrets.js:70]: I think optional chaining is what I miss most when not transpiling 😁

File: src/lib/secrets.js [hunk 6]
```diff
@@ -1,33 +1,149 @@
...
+
+  const newSecrets = formatSecrets(result)
+
+  return newSecrets
+}
 
 // eslint-disable-next-line promise/prefer-await-to-callbacks
-const withSecrets = (handler) => (event, context, callback) => {
-  const secrets = getSecrets()
+const withSecrets = (handler) => async (event, context, callback) => {
```
Comment (ascorbic) [inline src/lib/secrets.js:138]: The more I think about it, the more I think we should just use `getSecrets`, without the wrapper. It made sense when it was cheap and sync, but blocking on an HTTP request should really be an explicit thing.

**Review by ascorbic — COMMENTED** [review]
Looking good! Just a few comments.

File: src/lib/secrets.js (hunk 6 above)
Comment (dwwoelfel) [reply src/lib/secrets.js]: Removed `withSecrets` and all of the types in the `.d.ts` file that supported it.

**Pull request merged**